### PR TITLE
Add notebook to visualize E-GEOD-33245 trends

### PR DESCRIPTION
**Pull request by ajlee21**
This PR adds a notebook that looks at trends in Pseudomonas experiment, E-GEOD-33245.

The main changes can be found in:
1. `Viz_E_GEOD_33245_patterns.ipynb` which contains analysis results and visualizations
2. `process.py` which contains the supporting scripts for this notebook

The remaining changes are summary data files

### CODE REVIEW

File: generic_expression_patterns_modules/process.py [hunk 1]
```diff
@@ -289,7 +291,10 @@ def generate_summary_table(
         }
     )
     summary["Z score"] = (
-        summary["Test statistic (Real)"] - summary["Mean test statistic (simulated)"]
+        abs(
```
Comment (jjc2718) [inline generic_expression_patterns_modules/process.py:294]: Would it be useful to have a column in the summary table for the z-score, and then another one for the absolute value of the z-score? Or do you only expect to use the absolute value?

If the latter, I'd suggest naming the column something like `abs(Z score)`, just to remind yourself later on that these aren't the untransformed z-scores.

File: generic_expression_patterns_modules/process.py [hunk 2]
```diff
@@ -983,9 +956,418 @@ def concat_simulated_data_columns(local_dir, num_runs, project_id, data_type):
...
+            ]
+            .set_index("Gene ID")
+            .index
+        )
+    )
+    print(f"No. of specific DEGs using z-score: {len(degs_specific)}")
+
+    # Get predicted generic DEGs using z-score cutoff
+    # Z-score cutoff was found by calculating the score
+    # whose invnorm(0.05/5549)
```
Comment (jjc2718) [inline generic_expression_patterns_modules/process.py:1165]: I assume 5549 is the number of tests you're doing, is that right? So this is essentially a cutoff of p = 0.05, with a Bonferroni correction for 5549 tests?

File: generic_expression_patterns_modules/process.py [hunk 3]
```diff
@@ -983,9 +956,418 @@ def concat_simulated_data_columns(local_dir, num_runs, project_id, data_type):
...
+        degs_traditional,
+        degs_specific,
+        degs_generic,
+        degs_intersect,
+        degs_intersect_generic,
+        degs_diff,
+    )
+
+
+def plot_volcanos(
```
Comment (jjc2718) [inline generic_expression_patterns_modules/process.py:1231]: These functions are nice! So much more organized than my plotting/visualization code 😂

File: pseudomonas_analysis/nbconverted/Viz_E-GEOD-33245_patterns.py [hunk 4]
```diff
@@ -23,181 +25,245 @@
...
-# * Group 1v2: WT vs crc mutant
-# * Group 1v3: WT vs cbr mutant
+# ## Create dataframe to compare trends
+# We are going to merge data across different conditions. For example, we will merge `grp_1v2` and `grp_1v3` to use for plotting later in this notebook. Deb can look at these tables to find *things of interest* as we start looking into how to use our computational predictions of generic and specific genes. 
 # 
-# We can then compare the test statistic between these two groups above. We hope to see that,
-# * Genes 1v3  > 1v2 are genes that change more in 1v3 than 1v2 and we guess are specific to 1v3 (high z-score)
-# * Genes 1v3 < 1v2 are genes that change more in 1v2 than 1v3 and we guess are specific to 1v2 (high z-score)
-# * Genes on the 1v3 = 1v2 line are those genes that change equally in both and we guess are generic genes (low z-score)
+# She would like a dataframe with 1v2 and 1v3, 1v3 and 1v4.
```
Comment (jjc2718) [inline pseudomonas_analysis/nbconverted/Viz_E-GEOD-33245_patterns.py:40]: Are 1, 2, 3, 4, and 5 here just different datasets, or data collected in different conditions? Maybe you already have this information documented somewhere, I'm just forgetting.

File: pseudomonas_analysis/nbconverted/Viz_E-GEOD-33245_patterns.py [hunk 5]
```diff
@@ -23,181 +25,245 @@
...
 
-# Read data
-grp_1v2 = pd.read_csv(grp_1v2_file, sep="\t", header=0, index_col=0)
-grp_1v3 = pd.read_csv(grp_1v3_file, sep="\t", header=0, index_col=0)
-grp_1v4 = pd.read_csv(grp_1v4_file, sep="\t", header=0, index_col=0)
-
-grp_1v2_raw = pd.read_csv(grp_1v2_raw_file, sep="\t", header=0, index_col=0)
-grp_1v3_raw = pd.read_csv(grp_1v3_raw_file, sep="\t", header=0, index_col=0)
-grp_1v4_raw = pd.read_csv(grp_1v4_raw_file, sep="\t", header=0, index_col=0)
+process.plot_venn(degs_1v2_traditional, degs_1v2_specific, degs_1v2_generic)
```
Comment (jjc2718) [inline pseudomonas_analysis/nbconverted/Viz_E-GEOD-33245_patterns.py:159]: So would you interpret this as showing that the traditional DE results have more overlap with your specific genes than with generically DE genes? Is that what you expect here?

**Review by jjc2718 — APPROVED** [review]
Looks good! This is a neat analysis (and lots of pretty plots). I just have a few small questions.

**Review by georgiadoing — COMMENTED** [review]
This was my first review but, for whatever it's worth, I think it looked great! I made a couple spelling suggestions and some bigger picture comments but overall it was fun for me to read through (and understand what I could) and I am excited to see the new plots!

File: generic_expression_patterns_modules/process.py (hunk 1 above)
Comment (ajlee21) [reply generic_expression_patterns_modules/process.py:294]: We are using z-score to determine the "specificity" of genes so we care if they are different then the null, but I don't think we care about the direction for this. So I will add this text to the header I think that will definitely help future me!

File: generic_expression_patterns_modules/process.py (hunk 2 above)
Comment (ajlee21) [reply generic_expression_patterns_modules/process.py:1165]: yes. Here we are trying to get an equivalent z-value cutoff to match the adjusted p-value cutoff. I'll add a note here to clarify

File: pseudomonas_analysis/nbconverted/Viz_E-GEOD-33245_patterns.py (hunk 4 above)
Comment (ajlee21) [reply pseudomonas_analysis/nbconverted/Viz_E-GEOD-33245_patterns.py:40]: Good question. I will definitely add a comment here.

1,2,3,4,5 refer to different conditions/comparisons within the E-GEOD-33245 experiment. So 1 refers to samples comparing WT vs crc mutant.

File: pseudomonas_analysis/nbconverted/Viz_E-GEOD-33245_patterns.py (hunk 5 above)
Comment (ajlee21) [reply pseudomonas_analysis/nbconverted/Viz_E-GEOD-33245_patterns.py:159]: I would interpret these findings to mean that using traditional DE criteria you find a mix of generic and specific genes. And applying a z-score cutoff will supply researchers with a reasonable sized list of DEGs that are most relevant to the perturbagen in question that they can follow up with.

File: generic_expression_patterns_modules/process.py [hunk 6]
```diff
@@ -983,9 +956,418 @@ def concat_simulated_data_columns(local_dir, num_runs, project_id, data_type):
...
+        Either '1v2', '1v3', '1v4' or '1v5'
+    """
+    # Get DEGs using traditional criteria
+    degs_traditional = list(
+        (
+            merged_one_condition_df[
+                (merged_one_condition_df[f"Test statistic (Real)_grp_{condition}"] > 1)
+                & (
+                    merged_one_condition_df[f"Adj P-value (Real)_grp_{condition}"]
+                    < 0.05
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply generic_expression_patterns_modules/process.py:1130]: Thats a good point. I will add this change

File: pseudomonas_analysis/DEGs_E_GEOD_33245.tsv [hunk 7]
```diff
@@ -0,0 +1,595 @@
+	1v2 traditional + specific	1v2 specific only	1v2 traditional + generic	1v3 traditional + specific	1v3 specific only	1v3 traditional + generic	1v4 traditional + specific	1v4 specific only	1v4 traditional + generic	1v5 traditional + specific	1v5 specific only	1v5 traditional + generic
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply pseudomonas_analysis/DEGs_E_GEOD_33245.tsv:1]: This 1v5 is change in media, so WT in LB vs WT in BSM. I think that makes sense. But happy to hear more about what you think

File: generic_expression_patterns_modules/process.py [hunk 8]
```diff
@@ -983,9 +956,418 @@ def concat_simulated_data_columns(local_dir, num_runs, project_id, data_type):
...
+    and using z-score cutoff. Return different combinations of gene
+    lists.
+
+    Arguments
+    ---------
+    merged_one_condition_df: df
+        df of results for one of the E-GEOD-33245 conditions ('1v2', '1v3', '1v4' or '1v5')
+        returned from `merge_abs_raw_dfs`
+    condition: str
+        Condition from E-GEOD-33245 associated with 'merged_one_condition_df'.
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply generic_expression_patterns_modules/process.py:1120]: We discussed this offline, but the numbers will make it easier if we want to switch template experiments later. I have added what the different numbers translate to in the notebook for reference.

File: pseudomonas_analysis/nbconverted/Viz_E-GEOD-33245_patterns.py [hunk 9]
```diff
@@ -23,181 +25,245 @@
...
+                            '1v4 specific only': pd.Series(degs_1v4_diff),
+                            '1v4 traditional + generic': pd.Series(degs_1v4_intersect_generic),
+                            '1v5 traditional + specific': pd.Series(degs_1v5_intersect),
+                            '1v5 specific only': pd.Series(degs_1v5_diff),
+                            '1v5 traditional + generic': pd.Series(degs_1v5_intersect_generic)                            
+                           })
+degs_all_df.to_csv("DEGs_E_GEOD_33245.tsv", sep="\t")
+
+
+# **Takeaway:**
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply pseudomonas_analysis/nbconverted/Viz_E-GEOD-33245_patterns.py:271]: Yes! Very good point. I will add this as an issue for future work

File: generic_expression_patterns_modules/process.py [hunk 10]
```diff
@@ -983,9 +956,418 @@ def concat_simulated_data_columns(local_dir, num_runs, project_id, data_type):
...
+        },
+        palette=colors,
+        linewidth=0,
+        alpha=0.5,
+        ax=axes[2],
+    )
+
+    # Add labels
+    fig.suptitle(fig_title, fontsize=16)
+    axes[0].set_xlabel("log2 Fold Change")
```
[the remark this answers is not in the document]
Comment (ajlee21) [reply generic_expression_patterns_modules/process.py:1324]: It looks like I can! Good suggestion

**Pull request merged**